src/webparts/intranetSpaSpfx/screens/Home/CardNewsHome/index.tsx

```diff
@@ -0,0 +1,189 @@
+import { ReactElement, useEffect, useState, useCallback } from "react";
+import {
+  NewsContainer,
+  MainNews,
+  OtherNews,
+  NewsItem,
+  UserInfo,
+} from "./styles";
+import {
+  getNewsListPaginated,
+  TGetNewsListResponse,
+} from "../../../services/news.service";
+import { LikeViews } from "../../../components/LikeViews";
+import { useZustandStore } from "../../../store";
+import { getUser, TGetUserResponse } from "../../../services/user.service";
+import { Link } from "react-router-dom";
+import { formatDate } from "../../../utils/formatDate";
+import { motion } from "framer-motion";
+import { Spinner } from "../../../components/Spinner";
+
+const truncateText = (text: string, maxLength: number): string => {
+  if (text.length > maxLength) {
+    return text.slice(0, maxLength) + "...";
+  }
+  return text;
+};
+
+export const CardNewsHome = (): ReactElement => {
+  const { context } = useZustandStore();
+  const [news, setNews] = useState<TGetNewsListResponse[]>([]);
+  const [users, setUsers] = useState<Record<number, TGetUserResponse>>({});
+  const [loading, setLoading] = useState(true); // Estado de carregamento
+
+  const fetchNews = useCallback(async () => {
+    if (!context) return;
+
+    try {
+      const { data } = await getNewsListPaginated(context, 4);
+      setNews(data);
+
+      const userPromises = data.map((item) => getUser(context, item.AuthorId));
+      const userResults = await Promise.all(userPromises);
+
+      const usersMap = userResults.reduce(
+        (acc, user) => {
+          acc[user.Id] = user;
+          return acc;
+        },
+        {} as Record<number, TGetUserResponse>,
+      );
+
+      setUsers(usersMap);
+      setLoading(false);
+    } catch (error) {
+      console.error("Erro ao buscar as notícias ou usuários:", error);
+      setLoading(false);
+    }
+  }, [context]);
+
+  useEffect(() => {
+    fetchNews();
+  }, [fetchNews]);
+
+  if (loading) {
+    return (
+      <div
+        style={{
+          display: "flex",
+          justifyContent: "center",
+          alignItems: "center",
+          height: "100%",
+        }}
+      >
+        <Spinner color="#c02c2e" height={40} width={40} />
+      </div>
+    );
+  }
+
+  if (!news.length) {
+    return <p style={{ textAlign: "center" }}>Nenhuma notícia disponível.</p>;
+  }
+
+  const [mainNews, ...otherNews] = news.reverse();
+
+  return (
+    <NewsContainer>
+      {/* Notícia Principal */}
+      <motion.div
```

Comment:
{
(news && news.length)  ? 
(<>...</>) : 
(<p style={{ textAlign: "center" }}>Nenhuma notícia disponível.</p>)
}